base_layer/transaction_components/src/transaction_components/wallet_output.rs

```diff
@@ -517,8 +597,207 @@ impl WalletOutput {
             )
             .await?;
         self.metadata_signature = metadata_sig;
+        self.fix_hash();
+        Ok(())
+    }
+
+    fn fix_hash(&mut self) {
```

Comment:
```suggestion
    fn set_hash(&mut self) {
```
or ` fn update_hash(&mut self) {`